hazelcast/src/main/java/com/hazelcast/config/IndexType.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+/**
+ * Type of the index.
+ */
+public enum IndexType {
+    /** Sorted index. Can be used with equality and range predicates. */
+    SORTED(0),
+
+    /** Hash index. Can be used with equality predicates. */
+    HASH(1);
+
+    private final int id;
+
+    IndexType(int id) {
+        this.id = id;
+    }
+
+    /**
+     * Gets the ID for the given {@link IndexType}.
+     * <p>
+     * The reason this ID is used instead of an the ordinal value is that the ordinal value is more prone to changes
```

Comment:
Minor: `IndexType` is user-facing, so probably it's better to move this sentence into the method body since that information feels more like an implementation detail.

Reply:
Fixed in https://github.com/hazelcast/hazelcast/pull/15537/commits/2163cbdd5ac5290a80317aa92679e8ded7520b42